### PR TITLE
Add WordCloud plugin

### PR DESCRIPTION
**Pull request by kristw**
🏆 Enhancements

Add `WordCloudChartPlugin`

WIP: Need to resolve `reactify` dependency and add unit tests.

### CODE REVIEW

**Comment by xtinec** [comment]
Ported the word cloud plugin to TypeScript and added new tests for `transformProps`.  I think this PR is ready for review.

P.S. I think we could handle the `post-build` script for copying assets like `images/thumbnail.png` (and other build changes) in a separate PR.

File: packages/superset-ui-plugin-chart-word-cloud/types/external.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+declare module '*.png';
+declare module '@superset-ui/color';
```
Comment (kristw) [inline packages/superset-ui-plugin-chart-word-cloud/types/external.d.ts]: `color` is now in TS.

File: packages/superset-ui-plugin-chart-word-cloud/types/external.d.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+declare module '@superset-ui/translation';
```
Comment (kristw) [inline packages/superset-ui-plugin-chart-word-cloud/types/external.d.ts]: Has another PR converting translation to TS.

File: packages/superset-ui-chart/src/models/ChartPlugin.ts [hunk 3]
```diff
@@ -37,13 +37,13 @@ export interface ChartPluginConfig {
   loadChart?: PromiseOrValueLoader<Function>;
 }
 
-export default class ChartPlugin extends Plugin {
+export default class ChartPlugin<T extends FormData = FormData> extends Plugin {
```
Comment (xtinec) [inline packages/superset-ui-chart/src/models/ChartPlugin.ts]: What does the `= FormData` do?

File: packages/superset-ui-chart/src/models/ChartPlugin.ts [hunk 4]
```diff
@@ -13,20 +13,20 @@ const IDENTITY = (x: any) => x;
 type PromiseOrValue<T> = Promise<T> | T;
 type PromiseOrValueLoader<T> = () => PromiseOrValue<T> | PromiseOrValue<{ default: T }>;
 
-export type BuildQueryFunction = (formData: FormData) => QueryContext;
+export type BuildQueryFunction<T extends FormData> = (formData: T) => QueryContext;
```
Comment (williaster) [inline packages/superset-ui-chart/src/models/ChartPlugin.ts]: ❤️ this

File: packages/superset-ui-chart/src/models/ChartPlugin.ts (hunk 3 above)
Comment (williaster) [reply packages/superset-ui-chart/src/models/ChartPlugin.ts]: defaults to that generic?

Comment (kristw) [reply packages/superset-ui-chart/src/models/ChartPlugin.ts]: Yes, default to that generic so all the charts without specific `FormData` can fallback to it.

**Comment by kristw** [comment]
Umm, ci is throwing error that I don't see locally. Have to figure out what is going on.

File: packages/superset-ui-chart/src/models/ChartPlugin.ts (hunk 3 above)
Comment (xtinec) [reply packages/superset-ui-chart/src/models/ChartPlugin.ts]: Aha, got it. That makes sense.

**Pull request merged**